### PR TITLE
Prohibition on entering "+" and "-" characters in the 'Sum' field

### PR DESCRIPTION
**Pull request by iambrozjak**
* [Project ticket #746](https://github.com/ita-social-projects/ZeroWaste/issues/746)

## Code reviewers

- [ ] @loqimean 
- [x] @obniavko 

## Summary of issue
There shouldn't be an opportunity to enter any of those signs '-' and '+' in 'Sum' fields

## Summary of change
Added a ban on entering "+" and "-" characters in the 'Sum'  field when creating and editing a product
image
image
image

## CHECK LIST
- [x]  СI passed
- [x]  Сode coverage >=95%
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [x]  I've checked new feature as logged in and logged out user if needed
- [x]  PR meets all conventions

### CODE REVIEW

**Review by obniavko — APPROVED** [review]

File: app/views/account/products/partials/new/_form.html.erb [hunk 1]
```diff
@@ -10,6 +10,7 @@
             <%= f.simple_fields_for :prices, f.object.find_or_build_price_for_category(category) do |prices_form| %>
               <%= prices_form.input :sum, label: false, input_html: { placeholder: t(".form.sum"), data: { price_form_target: "priceInput" } } %>
               <%= prices_form.hidden_field :category_id, value: category.id %>
+              <div class="error-message" style="display: none; color: #9e1503;" data-price-form-target="errorMessage"><%= t(".form.error_sum") %></div>
```
Comment (loqimean) [inline app/views/account/products/partials/new/_form.html.erb:13]: інлайнові стилі, це зло

File: app/views/account/products/partials/edit/_form.html.erb [hunk 2]
```diff
@@ -11,6 +11,7 @@
                 <%= prices_form.input :sum, label: false, input_html: { placeholder: t('.form.sum'), data: { price_form_target: 'priceInput' } } %>
                 <%= prices_form.hidden_field :category_id, value: prices_form.object.category.id %>
                 <%= prices_form.hidden_field :_destroy, data: { price_form_target: 'hiddenField' } %>
+                <div class="error-message" style="display: none; color: #9e1503;" data-price-form-target="errorMessage"><%= t(".form.error_sum") %></div>
```
Comment (loqimean) [inline app/views/account/products/partials/edit/_form.html.erb:14]: аналогічно

File: app/javascript/controllers/price_form_controller.js [hunk 3]
```diff
@@ -37,4 +41,21 @@ export default class extends Controller {
       this.hiddenFieldTargets[i].value = !this.checkboxTargets[i].checked
     }
   }
+
+  validatePriceInput(event) {
+    const target = event.target;
+    const inputValue = target.value;
+    const regex = /^[^\+\-]+$/;
```
Comment (loqimean) [inline app/javascript/controllers/price_form_controller.js:48]: а чому не валідувати на тільки цифри? не розумію навіщо так ускладнювати

**Review by loqimean — CHANGES_REQUESTED** [review]

File: app/views/account/products/partials/new/_form.html.erb (hunk 1 above)
Comment (iambrozjak) [reply app/views/account/products/partials/new/_form.html.erb]: виправила)

File: app/javascript/controllers/price_form_controller.js (hunk 3 above)
Comment (iambrozjak) [reply app/javascript/controllers/price_form_controller.js]: змінила регулярний вираз)

**Review by loqimean — APPROVED** [review]

**Review by loqimean — APPROVED** [review]

**Pull request merged**